pages/api/roadmap.ts

```diff
@@ -1,102 +1,102 @@
 import type { NextApiRequest, NextApiResponse } from 'next';
-
-import _, { result } from 'lodash';
+import type {
+  GithubIssueDataWithGroup,
+  GithubIssueDataWithGroupAndChildren,
+  IssueData,
+  ParserGetChildrenResponse,
+  RoadmapApiResponse,
+  RoadmapApiResponseFailure,
+  RoadmapApiResponseSuccess
+} from '../../lib/types';
 
 import { getIssue } from '../../lib/backend/issue';
 import { getChildren, getConfig } from '../../lib/parser';
-import { IssueData, ParserGetChildrenResponse, RoadmapApiResponse, RoadmapApiResponseFailure, RoadmapApiResponseSuccess } from '../../lib/types';
 import { paramsFromUrl } from '../../lib/paramsFromUrl';
 import { errorManager } from '../../lib/backend/errorManager';
 import { checkForLabel } from '../../lib/backend/checkForLabel';
 
-const resolveChildren = (children: any[]): Promise<IssueData[]> => {
-  const resultArray: any = [];
-  let count = 0;
-
-  return new Promise((resolve, reject) => {
-    if (!_.isArray(children)) reject('Children is not an array. Is this a root issue?');
-    if (_.isArray(children) && children.length === 0) reject('Children array is empty. Is this a root issue?');
-
-    children.forEach((current) => {
-      getIssue(paramsFromUrl(current.html_url)).then((issueData) => {
-        checkForLabel(issueData);
-        resultArray.push({
-          ...issueData,
-          labels: issueData?.labels ?? [],
-          group: current.group,
-        });
-        count += 1;
-        if (count === children.length) {
-          resolve(resultArray);
-        }
-      });
-    });
-  });
+async function resolveChildren (children: ParserGetChildrenResponse[]): Promise<GithubIssueDataWithGroup[]> {
+  if (!Array.isArray(children)) {
+    throw new Error('Children is not an array. Is this a root issue?');
+  }
+
+  if (children.length === 0) {
+    throw new Error('Children array is empty. Is this a root issue?');
+  }
```

Comment:
+1. I was going to remove the duplicate isArray check but really wanted to stay as far away from modifying the code as I could when my focus wasn't explicitly focused on improving it.